doc/rst/language/spec/statements.rst

```diff
@@ -993,3 +996,227 @@ given by
+        var wrapper = new IntWrapper();
+        manage wrapper as val do val = 8;
+      }
+      manageIntWrapper();
+
+   produces the output
+
+   .. BLOCK-test-chapeloutput
+
+      entering
+      (x = 0)
+      leaving
+      (x = 8)
+
+   .. code-block:: bash
+
+      entering
+      (x = 0)
+      leaving
+      (x = 8)
+
+The ``enterThis()`` special method is called on the manager expression
+before executing the managed block (in the above example the manager
+expression is ``wrapper``). The method may return a type or value, or
+it may return ``void``.
+
+The resource returned by ``enterThis()`` can be captured by name so
+that it can be referred to within the scope of the managed block
+(in the above example the captured resource is ``val``).
+
+Capturing a returned resource is optional, and the syntax may be
+omitted. It is an error to try to capture a resource if
+``enterThis()`` returns ``void``.
+
+The storage of a captured resource may also be omitted, in which
+case it will be inferred from the return intent of the ``enterThis()``
+method (in the above example the storage of ``val`` is inferred
+to be ``ref``).
+
+Resource storage may also be specified explicitly.
+
+   *Example (manage2.chpl)*.
+
+
+
+   .. code-block:: chapel
+
+      record IntWrapper {
+        var x: int;
+      }
+
+      proc IntWrapper.enterThis() ref: int {
+        writeln('entering');
+        writeln(this);
+        return this.x;
+      }
+
+      proc IntWrapper.leaveThis(in error: owned Error?) throws {
+        if error then throw error;
+        writeln('leaving');
+        writeln(this);
+      }
+
+      proc manageIntWrapper() {
+        var wrapper = new IntWrapper();
+        manage wrapper as var val {
+          val = 8;
+        }
+      }
+      manageIntWrapper();
+
+   produces the output
+
+   .. BLOCK-test-chapeloutput
+
+      entering
+      (x = 0)
+      leaving
+      (x = 0)
+
+   .. code-block:: bash
+
+      entering
+      (x = 0)
+      leaving
+      (x = 0)
+
+Because the storage of ``val`` was specified as ``var``, the integer
+field of ``wrapper`` was not modified even though ``enterThis()``
+returns by ``ref``.
+
+.. note::
+
+  *Open issue:*
+
+    The ``enterThis()`` special method does not currently support the
+    use of return intent overloading (see :ref:`Return_Intent_Overloads`)
+    when the storage of a resource is omitted. Adding such support would
+    require additional disambiguation rules, and the value of doing so
+    is unclear at this time.
+
+Participating types must also define the ``leaveThis()`` method,
+which is called implicitly when the scope of the managed block
+is exited.
+
+The ``leaveThis()`` method takes an ``Error?`` by ``in`` intent. If
+the error is not ``nil``,  it may be handled within the method. It
+can also be propagated by annotating ``leaveThis()`` with the
+``throws`` tag and throwing the error.
+
+Manager expressions may also be nested within a manage statement.
```

Comment:
I tend to think of nesting as:
```chapel
manage wrapper1 as val1 {
  manage wrapper2 as val2 {
    ...
  }
}
```
It seems like an implementation detail that the code you've written counts as nested, though I think the ordering of entering and leaving is still important to convey.  I think I'd use similar wording to [multiple variable declarations](https://chapel-lang.org/docs/1.26/language/spec/variables.html#multiple-variable-declarations)

Reply:
Semantically the "nesting" is equivalent to the code you've written here. I'll give that link a look. Maybe something like "multiple manager expressions may be present within a single manage statement".